packages/browser/src/core/analytics/index.ts

```diff
@@ -534,6 +534,24 @@ export class Analytics
   async ready(
     callback: Function = (res: Promise<unknown>[]): Promise<unknown>[] => res
   ): Promise<unknown> {
+    if (this.options.ignorePluginReadyError) {
+      return Promise.allSettled(
+        this.queue.plugins.map((i) =>
+          i.ready
+            ? i.ready().catch(this.options.onPluginReadyError)
+            : Promise.resolve()
+        )
+      ).then((results) => {
+        const res = results.map((result) => {
+          if (result.status === 'fulfilled') {
+            return result.value
+          }
+        })
+        callback(res)
+        return res
+      })
+    }
+
     return Promise.all(
```

Comment:
Doing an allSettled and then an .all here like duplicate work -- Can we refactor to just use allSettled?

Reply:
Will do!

Reply:
[6b0d9fe](https://github.com/segmentio/analytics-next/pull/1282/commits/6b0d9feb72c97a6f5d1d8152ec861f79ab1f8b93)